### PR TITLE
feat: rework libem serve with support for oauth

### PR DESCRIPTION
**Pull request by daiwaid**
#### What this PR does / why we need it:
- Reworked libem serve with OAuth
- Accessible at https://serve.libem.org

#### Which issue(s) this PR addresses (if any):

Addresses #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -14,10 +14,6 @@ numpy
 pandas
 tabulate
 bs4
-flask
```
Comment (zenodflow) [inline requirements.txt:17]: Just to be sure - are there any modules outside serve/ rely on these dependencies?

Btw., we should consider separate the libem / arena / serve/ benchmark dependencies with individual requirements.txt.

File: serve/libem.yaml [hunk 2]
```diff
@@ -14,18 +14,23 @@ spec:
     spec:
       containers:
       - name: libem-web-service
-        image: silveryfu/libem-serve:latest
+        image: daiwaid/libem-serve:latest
```
Comment (zenodflow) [inline serve/libem.yaml:17]: Make sure the repo is public on docker registry

File: serve/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+uvicorn
+requests
+fastapi
+itsdangerous
+python-dotenv
+fastapi-login
+authlib
+mysql-connector-python
+https://github.com/abcsys/libem/archive/refs/heads/main.zip
```
[the remark this answers is not in the document]
Comment (zenodflow) [reply serve/requirements.txt:9]: Btw., I see you already do the requirements separation here! Nice.

In future, we should reflect this in the main Makefile, e.g, `make install-serve`, `make install-arena` `make install-benchmark` etc.

**Review by zenodflow — COMMENTED** [review]
Great update. I wonder if it is possible to add a simple, local e2e test case for the serve? E.g., start the serve and return a "guess" result.

Also, perhaps copy over the README in arena to here (tailor it a bit for serve)?

File: serve/requirements.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (daiwaid) [reply serve/requirements.txt:9]: This is to add the latest version of libem package without needing to use git.

File: requirements.txt (hunk 1 above)
Comment (daiwaid) [reply requirements.txt:17]: Yeah I added individual requirements.txt for arena and serve and removed those packages from the main file. It's all packages used for API so it shouldn't be used anywhere else.

File: serve/libem.yaml (hunk 2 above)
Comment (daiwaid) [reply serve/libem.yaml:17]: Should we make a libem account?

File: requirements.txt (hunk 1 above)
Comment (zenodflow) [reply requirements.txt:17]: Great!

**Review by zenodflow — APPROVED** [review]

**Pull request merged**